pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -148,13 +142,16 @@ def check_metrics(self, logs):
 
         return True
 
-    def on_train_begin(self, logs=None):
+    def on_train_begin(self):
         # Allow instances to be re-used
         self.wait = 0
         self.stopped_epoch = 0
         self.best = np.Inf if self.monitor_op == np.less else -np.Inf
 
-    def on_epoch_end(self, epoch, logs=None):
+    def on_epoch_end(self):
+        assert self._trainer is not None, _no_trainer_error_msg
```

Comment:
I would move this assert to a parent class and just at the beginning call parent method ?

Reply:
I am not sure about this. Theoretically there may be some calls or even entire callbacks which don't use trainer at all. So there will be no need in this assert.

Reply:
true... lets keep it for now